backend/server/rhnRepository.py

```diff
@@ -235,30 +235,28 @@ def _repodata_taskomatic(self, file_name):
 
         content_type = "application/x-gzip"
 
-        if file_name in ["repomd.xml", "comps.xml", "products.xml"]:
+        if file_name.endswith(".xml"):
             content_type = "text/xml"
         elif file_name in ["repomd.xml.asc", "repomd.xml.key"]:
             content_type = "text/plain"
-        elif file_name not in ["primary.xml.gz", "other.xml.gz",
-                               "filelists.xml.gz", "updateinfo.xml.gz", "Packages.gz", "modules.yaml",
-                               "InRelease", "Release", "Release.gpg",
-                               "susedata.xml.gz"]:
-            log_debug(2, "Unknown repomd file requested: %s" % file_name)
-            raise rhnFault(6)
-
-        # XXX this won't be repconned or CDNd
+        elif file_name.endswith(".yaml"):
+             content_type = "text/yaml"
+        file_path = "%s/%s/%s" % (CFG.REPOMD_PATH_PREFIX, self.channelName, file_name)
         if file_name in ["comps.xml", "modules.yaml"]:
+            # without checksum in the filename, they are only available in the old style
             return self._repodata_python(file_name)
+        elif not os.path.exists(CFG.REPOMD_CACHE_MOUNT_POINT + "/" + file_path):
```

Comment:
I believe that this is the saner way. The only question in my eyes is now if the filename is already sanitized because if not then this is the perfect entry point of what we just had with Cobbler.

```suggestion
        elif not os.path.exists(os.path.join(CFG.REPOMD_CACHE_MOUNT_POINT, file_path)):
```